server/sonar-server/src/test/java/org/sonar/server/component/NewComponentTest.java

```diff
@@ -42,11 +42,11 @@ public void fail_when_key_is_longer_than_400_characters() {
     expectedException.expect(IllegalArgumentException.class);
     expectedException.expectMessage("Component key length (401) is longer than the maximum authorized (400)");
 
-    NewComponent.create(randomAlphabetic(401), "name");
+    NewComponent.create(repeat("a", 400 + 1), "name");
   }
 
   @Test
-  public void fail_when_name_is_null() {
+  public void fail_with_functional_exception_when_name_is_null() {
```

Comment:
not everywhere, just null testing (it's a mistake regarding the 33 line). I use checkArgument and not requireNonNull because I want an IllegalArgumentException and not an NPE. Http code is 4XX and not 5XX when handled by the WS engine